common/src/main/java/com/wynntils/models/seaskipper/SeaskipperModel.java

```diff
@@ -0,0 +1,202 @@
+
+        availableNames.add(destination);
+    }
+
+    private void addDestination(SeaskipperDestinationItem destination, int slot) {
+        destinations.put(destination, slot);
+    }
+
+    public Map<SeaskipperDestinationItem, Integer> getDestinations() {
+        return Collections.unmodifiableMap(destinations);
+    }
+
+    public List<SeaskipperPoi> getAvailableDestinations() {
+        return Collections.unmodifiableList(availableDestinations);
+    }
+
+    public List<SeaskipperPoi> getSeaskipperPois() {
+        return Collections.unmodifiableList(seaskipperPois);
+    }
+
+    private void setBoatSlot(int boatSlot) {
+        this.boatSlot = boatSlot;
+    }
+
+    public void purchaseBoat() {
+        clickSlot(boatSlot);
+    }
+
+    public SeaskipperPoi getCurrentPoi() {
+        return currentPoi;
+    }
+
+    public void buyPass(String destinationToTravelTo) {
+        int passSlot = -1;
+
+        for (Map.Entry<SeaskipperDestinationItem, Integer> entry :
+                Models.Seaskipper.getDestinations().entrySet()) {
+            String destination = entry.getKey().getDestination();
+
+            if (destination.equals(destinationToTravelTo)) {
+                passSlot = entry.getValue();
+                break;
+            }
+        }
+
+        if (passSlot == -1) {
+            return;
+        }
+
+        clickSlot(passSlot);
+    }
+
+    private void clickSlot(int slot) {
+        ContainerUtils.clickOnSlot(
+                slot,
+                actualSeaskipperScreen.getMenu().containerId,
+                GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                actualSeaskipperScreen.getMenu().getItems());
+    }
+
+    private void loadSeaskipperPois() {
+        seaskipperPois.clear();
+        availableNames.clear();
+        availableDestinations.clear();
+        destinations.clear();
+
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_SEASKIPPER_LOCATIONS);
+        dl.handleReader(reader -> {
+            Type type = new TypeToken<ArrayList<SeaskipperProfile>>() {}.getType();
+            List<SeaskipperProfile> seaskipperProfiles = WynntilsMod.GSON.fromJson(reader, type);
+
+            for (SeaskipperProfile profile : seaskipperProfiles) {
+                seaskipperPois.add(new SeaskipperPoi(
+                        profile.destination,
+                        profile.combatLevel,
+                        profile.startX,
+                        profile.startZ,
+                        profile.endX,
+                        profile.endZ));
+            }
+
+            for (SeaskipperPoi poi : seaskipperPois) {
+                if (poi.isPlayerInside()) {
+                    currentPoi = poi;
+                    availableDestinations.add(currentPoi);
+                    break;
+                }
+            }
+
+            setAvailableDestinations();
+        });
+    }
+
+    private void setAvailableDestinations() {
```

Comment:
This is a bit confused. You are doing the same work twice, it seems. You don't need "availableNames". 

AvailableDestinations gets populated by the setSlot event. It seems you also want to add the players current position as a destination. You can do that in `onMenuOpened`.